src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -35,12 +39,28 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnBrokerChanged(object sender, EventArgs e) {
+        private void OnRSessionInitialized(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
```

Comment:
> Also, this is the event when connection should turn green and the progress dialog should go away.

No. If we want to turn green only when Interactive Window is visible, we need to track both broker and window. Since window is on UI thread, there can be a big gap between the moment when RHost is ready for user input and Interactive Window has completed its initialization (which is what you actually see in #2736).